src/common/utxobased/engine/types.ts

```diff
@@ -1,7 +1,20 @@
+import { asArray, asBoolean, asObject, asOptional, asString } from 'cleaners'
 import { EdgeSpendInfo } from 'edge-core-js/types'
 
 import { Input, Output } from '../keymanager/utxopicker/types'
 
+// Normalizes a URL to always be a websocket URL
+const asWebsocketUrl = (raw: unknown): string => {
```

Comment:
One concern I have about doing this here is that it means the engine could become un-bootable if the user provides invalid user settings. We may want to either:

1. Go back to a plain `asString` and just let the connection fail if it's not a valid URL. Server score would go to 0, and all would be well.
2. Filter the invalid entries out, either here in the cleaner or later when this is used.

I prefer option 1, since the normalization that's happening here is over-agressive and kinda daft.